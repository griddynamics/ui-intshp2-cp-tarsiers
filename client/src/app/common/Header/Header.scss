@import '../../../styles/variables';
@import '../../../styles/mixin';

a {
  &:not(div):hover {
    color: $color-main !important;
  }
}

.header {
  width: 100%;
  padding-bottom: 40px;
  background-color: $color-white;
  font-family: $main-font-regular;
  position: relative;

  &-top {
    padding: 10px;
    padding-top: 30px;
    @include d-flex();

    .far,
    .fab {
      color: $color-main;
      padding-right: 10px;
    }

    .fa-skype {
      padding-left: 10px;
    }

    .icons-right {
      margin-left: auto;

      ul {
        @include d-flex();
      }

      i {
        padding: 0 10px;
        color: inherit;
      }
    }

    span {
      padding-right: 5px;
    }
  }

  &-main {
    padding-top: 20px;
    @include d-flex();
  }

  &-links {
    margin-top: 10px;
    @include d-flex(space-around);
  }

  &-logo {
    img {
      padding: 0 5px;
      height: auto;
      width: 120px;
    }
  }

  &-nav-right {
    & > .header-links {
      justify-content: flex-end;
      align-items: center;
      li {
        padding: 0 20px;
      }
    }
  }
}

.nav-main {
  min-width: 60%;
}

.line-between {
  border-bottom: 1px solid $color-light-2;
}

.navbar {
  @include d-flex();
}

.menu {
  display: none;
}

@media screen and (max-width: $media-tablet) {
  .header {
    &-links > li {
      padding: 5px 10px;
    }

    &-top {
      @include d-flex() {
        flex-flow: column wrap;
        align-items: center;
      }

      .icons-right {
        margin: 0;
        font-size: 20px;
        padding: 5px 0;
      }

      i {
        padding-top: 5px;
      }
    }

    &-nav-right {
      text-align: right;
      width: 100%;
    }
  }

  .menu {
    display: inline-block;
    vertical-align: baseline;
    font-size: 18px;
    cursor: pointer;
    outline: none;
    border: none;
    background: none;
    margin-right: 10px;
    margin-top: -3px;
  }

  .nav-main {
    display: none;
  }

  .nav-toggle {
    display: flex;
    position: absolute;
    top: 90%;
    left: 0;
    right: 0;
    z-index: 3;

    .header-links {
      text-align: center;
      background-color: $color-light-1;
      padding: 10px;
      margin: 0;
      width: 100%;
      @include d-flex() {
        flex-flow: column nowrap;
      }
    }
  }
}

.cart-icon {
  @include d-flex(unset, column);
  align-items: center;
  position: relative;

  &__quantity {
    position: absolute;
    bottom: 65%;
    color: $color-dark;
    font-size: 13px;
  }
}
