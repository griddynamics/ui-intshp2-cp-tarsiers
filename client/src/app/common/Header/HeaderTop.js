import React from 'react';
import { ExternalLink } from '../Links';

const HeaderTop = props => {
  const renderLinks = list =>
    list.map(link => {
      const { id, href, icon } = link;

      return (
        <li key={id}>
          <ExternalLink href={href}>
            <i className={icon} />
          </ExternalLink>
        </li>
      );
    });

  const { links, contacts } = props;
  const socialLinks = <ul>{renderLinks(links)}</ul>;

  return (
    <div className="line-between">
      <div className="header-top container">
        <a href={`mailto:${contacts.email}`}>
          <i className="far fa-envelope" />
          {contacts.email}
        </a>
        <a href={`skype:${contacts.phone}`}>
          <i className="fab fa-skype" />
          {contacts.phone}
        </a>
        <div className="icons-right">{socialLinks}</div>
      </div>
    </div>
  );
};

export default HeaderTop;
