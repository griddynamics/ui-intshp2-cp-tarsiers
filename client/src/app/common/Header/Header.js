import React from 'react';
import './Header.scss';
import HeaderTop from './HeaderTop';
import HeaderMain from './HeaderMain';
import appConfig from '../../../config/appConfig';

const Header = props => {
  const { toggleHFVisibility, cart } = props;
  const visible = toggleHFVisibility.value;
  const { contacts, links, pages, options } = appConfig.header;

  return (
    visible && (
      <header className="header">
        <HeaderTop contacts={contacts} links={links} />
        <HeaderMain pages={pages} options={options} cart={cart} />
      </header>
    )
  );
};

Header.defaultProps = {
  cart: {
    value: 0
  }
};

export default Header;
