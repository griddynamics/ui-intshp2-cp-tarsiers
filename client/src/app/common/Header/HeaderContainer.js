import { connect } from 'react-redux';
import Header from './Header';

const mapStateToProps = state => ({
  cart: state.cart,
  toggleHFVisibility: state.toggleHFVisibility
});

export default connect(mapStateToProps)(Header);
