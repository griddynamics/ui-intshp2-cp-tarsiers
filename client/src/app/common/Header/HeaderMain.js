import React, { Component } from 'react';
import { RouterLink } from '../Links';

import './Header.scss';
import logo from '../../../assets/logo.png';

class HeaderMain extends Component {
  state = { isHidden: true };

  navToggle = () => {
    const { innerWidth } = window;

    if (innerWidth < 768) {
      this.setState(prevState => ({ isHidden: !prevState.isHidden }));
    }
  };

  renderLinks = list => {
    const { cart } = this.props;
    const cartValue = cart.productsInCart.length;

    return list.map(link => {
      const { id, icon, href, desc, value } = link;
      const headerIcon =
        desc === 'cart' ? (
          <div className="cart-icon">
            <span className="cart-icon__quantity">{cartValue}</span>
            <i className={icon} />
          </div>
        ) : (
          <i className={icon} />
        );

      return (
        <li key={id} className="navbar-link">
          <RouterLink href={href} callback={this.navToggle}>
            {icon ? headerIcon : value}
          </RouterLink>
        </li>
      );
    });
  };

  render() {
    const { isHidden } = this.state;
    const { pages, options } = this.props;
    const navItems = (
      <ul className="header-links">{this.renderLinks(pages)}</ul>
    );
    const userOptions = (
      <ul className="header-links">
        {this.renderLinks(options)}
        <button type="button" className="menu" onClick={this.navToggle}>
          <i className="fas fa-bars" />
        </button>
      </ul>
    );

    return (
      <div className="header-main container">
        <div className="header-logo col-2">
          <RouterLink href="/">
            <img src={logo} alt="logo" />
          </RouterLink>
        </div>
        <nav className="navbar col-10">
          <div className={!isHidden ? 'nav-toggle' : 'nav-main'}>
            {navItems}
          </div>
          <div className="header-nav-right col-3">{userOptions}</div>
        </nav>
      </div>
    );
  }
}

export default HeaderMain;
