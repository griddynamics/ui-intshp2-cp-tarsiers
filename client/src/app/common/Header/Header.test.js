import React from 'react';
import { shallow, mount } from 'enzyme';
import Header from './Header';

jest.mock('react-router-dom');
describe('<Header />', () => {
  let props;

  beforeEach(() => {
    props = {
      toggleHFVisibility: { value: true },
      cart: { productsInCart: [] }
    };
  });

  it('should match its snapshot', () => {
    const wrapper = shallow(<Header {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should able to render all child components', () => {
    const wrapper = mount(<Header {...props} />);

    expect(wrapper).toBeDefined();
  });

  it('should be able to render based on props', () => {
    const wrapper = mount(<Header {...props} />);

    expect(wrapper.prop('toggleHFVisibility').value).toBe(true);
    wrapper.setProps({ toggleHFVisibility: { value: false } });
    expect(wrapper.prop('toggleHFVisibility').value).toBe(false);
  });
});
