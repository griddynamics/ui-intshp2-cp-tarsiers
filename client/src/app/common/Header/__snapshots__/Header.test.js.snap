// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match its snapshot 1`] = `
<header
  className="header"
>
  <HeaderTop
    contacts={
      Object {
        "email": "info@shopy.com",
        "phone": "453 - 5553 - 996",
      }
    }
    links={
      Array [
        Object {
          "href": "https://twitter.com/griddynamics",
          "icon": "fab fa-twitter",
          "id": 1,
        },
        Object {
          "href": "https://www.linkedin.com/company/grid-dynamics",
          "icon": "fab fa-linkedin-in",
          "id": 2,
        },
        Object {
          "href": "https://www.instagram.com/griddynamics_ua/",
          "icon": "fab fa-instagram",
          "id": 3,
        },
        Object {
          "href": "https://www.facebook.com/griddynamics",
          "icon": "fab fa-facebook-f",
          "id": 4,
        },
        Object {
          "href": "https://plus.google.com/115302417170674279390",
          "icon": "fab fa-google-plus-g",
          "id": 5,
        },
      ]
    }
  />
  <HeaderMain
    cart={
      Object {
        "productsInCart": Array [],
      }
    }
    options={
      Array [
        Object {
          "desc": "search",
          "href": "/",
          "icon": "fa fa-search",
          "id": 11,
        },
        Object {
          "desc": "user",
          "href": "/",
          "icon": "fa fa-user",
          "id": 12,
        },
        Object {
          "desc": "cart",
          "href": "/cart",
          "icon": "fas fa-shopping-basket",
          "id": 13,
        },
      ]
    }
    pages={
      Array [
        Object {
          "href": "/",
          "id": 6,
          "value": "Home",
        },
        Object {
          "href": "/products",
          "id": 7,
          "value": "Products",
        },
        Object {
          "href": "/hot-deals",
          "id": 8,
          "value": "Hot Deals",
        },
        Object {
          "href": "/about",
          "id": 9,
          "value": "About",
        },
        Object {
          "href": "/contacts",
          "id": 10,
          "value": "Contact",
        },
      ]
    }
  />
</header>
`;
